File: 프로그래머스/lv2/70129. 이진 변환 반복하기/이진 변환 반복하기.py
def solution(s):
    binary_transform_count = 0  
    removed_zero_count = 0  

    while s != "1":  
        new_s = ""
        for char in s:
            if char == "1":
                new_s += char
            else:
                removed_zero_count += 1

        binary_transform_count += 1
        s = bin(len(new_s))[2:] 

    return [binary_transform_count, removed_zero_count]

File: 프로그래머스/2/70129. 이진 변환 반복하기/이진 변환 반복하기.java
public class Solution {
    public int[] solution(String s) {
        int binaryTransformCount = 0;
        int removedZeroCount = 0;

        while (!s.equals("1")) {
            StringBuilder newS = new StringBuilder();
            for (char c : s.toCharArray()) {
                if (c == '1') {
                    newS.append(c);
                } else {
                    removedZeroCount++;
                }
            }

            binaryTransformCount++;
            s = Integer.toBinaryString(newS.length());
        }

        int[] result = { binaryTransformCount, removedZeroCount };
        return result;
    }

    public static void main(String[] args) {
        Solution solution = new Solution();
        String s = "110010101001";
        int[] result = solution.solution(s);
        System.out.println("Binary Transform Count: " + result[0]);
        System.out.println("Removed Zero Count: " + result[1]);
    }
}
